python/cudf/cudf/core/column/timedelta.py

```diff
@@ -250,6 +250,34 @@ def as_numerical(self) -> "cudf.core.column.NumericalColumn":
     def time_unit(self) -> str:
         return self._time_unit
 
+    def _preprocess_column_for_repr(self):
```

Comment:
Should this logic be going into `as_string_column()`?

Reply:
We can move this logic to `as_string_column` but I expect some added drag for `datetime`/`time-delta` to `string` conversions because of `any()` checks. I went ahead and added only for repr code because that is not a perf critical code.

Here is how the numbers look like:
```python
In [1]: import cudf

In [2]: s = cudf.Series([123200000, 42300000, 342340000], dtype='datetime64[ns]')

In [3]: s = s.repeat(10000000)

In [4]: s
Out[4]: 
0    1970-01-01 00:00:00.123200
0    1970-01-01 00:00:00.123200
0    1970-01-01 00:00:00.123200
0    1970-01-01 00:00:00.123200
0    1970-01-01 00:00:00.123200
                ...            
2    1970-01-01 00:00:00.342340
2    1970-01-01 00:00:00.342340
2    1970-01-01 00:00:00.342340
2    1970-01-01 00:00:00.342340
2    1970-01-01 00:00:00.342340
Length: 30000000, dtype: datetime64[ns]

# After moving `.any()` checks to `as_string_column`
In [5]: %timeit f = s.astype('str')
80.8 ms ± 170 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


# `branch-22.10`:
In [4]: %timeit f = s.astype('str')
56.3 ms ± 82.8 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

It also feels like we should move it to `as_string_column` for correctness and parity:

```python
In [17]: s = pd.Series([123200000, 42300000, 342340000], dtype='datetime64[ns]')

In [18]: s
Out[18]: 
0   1970-01-01 00:00:00.123200
1   1970-01-01 00:00:00.042300
2   1970-01-01 00:00:00.342340
dtype: datetime64[ns]

In [19]: gs = cudf.from_pandas(s)

In [20]: gs
Out[20]: 
0    1970-01-01 00:00:00.123200
1    1970-01-01 00:00:00.042300
2    1970-01-01 00:00:00.342340
dtype: datetime64[ns]


In [21]: s.astype('str') #Pandas omit's the nanoseconds
Out[21]: 
0    1970-01-01 00:00:00.123200
1    1970-01-01 00:00:00.042300
2    1970-01-01 00:00:00.342340
dtype: object

In [22]: gs.astype('str') # Whereas we don't omit and generate full timestamps
Out[22]: 
0    1970-01-01 00:00:00.123200000
1    1970-01-01 00:00:00.042300000
2    1970-01-01 00:00:00.342340000
dtype: object
```

So I'm open to either one of these two approaches, if need perf we shouldn't have it in `as_string_column`, if want close parity we will have to move it to `as_string_column`, let me know what you think about it?

Reply:
Thanks for the explanation. I do think that `dt_series.astype("str")` and `repr(dt_series)` should return timestamps in the same format, but at the same time, it would be nice to have an internal fast path for converting timestamps to strings without worrying about Pandas behavior.

I wonder if `as_string_column()` could provide that fastpath via a kwarg? For example, `as_string_column(format=None)` would use the `has_s, has_ms` etc., checks to determine `format`, while `as_string_column(format=format)` would not.

Reply:
> Thanks for the explanation. I do think that `dt_series.astype("str")` and `repr(dt_series)` should return timestamps in the same format. At the same time, it would be nice to have an internal fast path for converting timestamps to strings without worrying about Pandas behavior.
> 
> I wonder if `as_string_column()` could provide that fastpath via a kwarg? For example, `as_string_column(format=None)` would use the `has_s, has_ms` etc., checks to determine `format`, while `as_string_column(format=format)` would not.

Yup, the above numbers were using the same approach. I'll make the change 👍

Reply:
Updated the approach, @shwina this should be ready for review.